### PR TITLE
add health probe for karmada-search

### PR DESCRIPTION
**Pull request by AllenZMC**
**What type of PR is this?**
/kind feature
/kind cleanup

**What this PR does / why we need it**:
add health probe for karmada-search
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/karmada-search/app/options/options.go [hunk 1]
```diff
@@ -122,3 +131,14 @@ func (o *Options) Config() (*search.Config, error) {
 	}
 	return config, nil
 }
+
+func serveHealthzAndMetrics(address string) {
```
Comment (Poor12) [inline cmd/karmada-search/app/options/options.go:135]: Karmada-search is implemented on genericapiserver. It has implemented the healthz probe, default port is 443. You can based on `vendor/k8s.io/apiserver/pkg/server/config.go`.

Comment (AllenZMC) [reply cmd/karmada-search/app/options/options.go]: OK

**Comment by RainbowMango** [comment]
@XiShanYongYe-Chang Have you noticed the failing test?
https://github.com/karmada-io/karmada/runs/6996314058?check_suite_focus=true

I'll re-test it after your confirmation.

**Comment by AllenZMC** [comment]
> The CI failed for
> 
> ```
> Readiness probe failed: Get "[https://10.244.0.9:443/readyz](https://10.244.0.9/readyz)": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
> ```
> 
> It may not work when we retest the CI.

Do you know why it doesn't work?

**Comment by XiShanYongYe-Chang** [comment]
> Do you know why it doesn't work?

Sorry, I have no idea for now.

**Comment by XiShanYongYe-Chang** [comment]
It looks like a deadlock occurred: `apiservices: v1alpha1.search.karmada.io` wait for `karmada-search` pod to running, and the `karmada-search` component need to list/watch `*v1alpha1.ResourceRegistry`.

**Comment by XiShanYongYe-Chang** [comment]
Hi @AllenZMC, we can consider only adding `livenessProbe`.

**Comment by AllenZMC** [comment]
> Hi @AllenZMC, we can consider only adding `livenessProbe`.

Thanks， I will try it.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**